From 20b181d7beed9ecf2892f660265077955fee291c Mon Sep 17 00:00:00 2001
From: "Apekshit(Appy) Sharma" <appy@cloudera.com>
Date: Tue, 26 May 2015 20:11:49 -0700
Subject: [PATCH] HBASE-13783 Improve error message "Could not seek
 StoreFileScanner" to indicate that issue could be a bad disk. (Apekshit)

---
 .../java/org/apache/hadoop/hbase/regionserver/StoreFileScanner.java   | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreFileScanner.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreFileScanner.java
index 42a378d..a2156c9 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreFileScanner.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreFileScanner.java
@@ -175,7 +175,9 @@ public class StoreFileScanner implements KeyValueScanner {
     } catch (FileNotFoundException e) {
       throw e;
     } catch (IOException ioe) {
-      throw new IOException("Could not seek " + this + " to key " + key, ioe);
+      throw new IOException("Could not seek " + this + " to key " + key
+          + "\nThis is frequently a side effect of a bad disk on the node where the region is"
+          + " hosted. Moving the region manually usually resolves the issue.", ioe);
     }
   }
 
-- 
2.3.2 (Apple Git-55)

